### PR TITLE
[DNM] custom modal and more flexible tabs

### PR DESCRIPTION
**Pull request by geoffreymuchai**

### CODE REVIEW

File: src/hatti/views/table.cljs [hunk 1]
```diff
@@ -303,6 +303,21 @@
...
+        main-navigation-height (-> "#dataview-menu"
+                                   js/document.querySelector
+                                   .-clientHeight)
+        tab-navigation-height (-> ".tab-bar"
+                                  js/document.querySelector
+                                  .-clientHeight)]
+    (- body-height
+       (+ main-navigation-height
+          tab-navigation-height))))
+
```
Comment (okal) [inline src/hatti/views/table.cljs]: Still wondering how these got overwritten on this branch. I'd done this here: f150339fcd4a9d4052c01dfac9e6863baa85162d.

File: src/hatti/views/table.cljs [hunk 2]
```diff
@@ -331,19 +346,25 @@
...
-                                  {:view :table})})
+         [:div {:class "table-view"
+                :style (when prevent-scrolling-in-table-view?
+                         {:height (get-table-view-height)
+                          :overflow "hidden"})}
+          (when (:data submission-clicked)
+            (om/build submission-view
+                      (with-info submission-clicked)
+                      {:opts (merge (select-keys opts #{:delete-record! :role})
+                                    {:view :table})}))
```
Comment (okal) [inline src/hatti/views/table.cljs]: And this here: 96ae7c6867661821c94b7f32ff12c2553f9e4380

**Comment by okal** [comment]
There was also functionality to resize when the window size chages: b89ed034131b5fa1111090c754e828bc6874a901. You should probably restore that as well.

**Comment by okal** [comment]
6796799c8a6695798d48907be87fd9dc7576c9ac - added additional offsets to accommodate small screens better.

**Comment by okal** [comment]
All that was needed was to revert e2f92e938863f17162f24360df5e131bebccd0b2. I suspect that commit was added here due to a rebase/merge with master or something. It shouldn't be on the dfid-syria branch at all.

**Comment by okal** [comment]
None of this should go into master, BTW. It's only for dfid-syria.

**Comment by okal** [comment]
The `dfid-syria` branch is meant to keep `master` clean. These changes aren't meant to be in` master` at all. I made a mistake and added them when I filed the initial PR with the changes. That's why e2f92e938863f17162f24360df5e131bebccd0b2 needed to be done. So I'd suggest:

```sh
git checkout dfid-syria
git checkout -b restore-table-changes
git revert e2f92e938863f17162f24360df5e131bebccd0b2
git push -u
```
Then file a PR from `restore-table-changes` against `dfid-syria`, not `master`.

File: src/hatti/views.cljs [hunk 3]
```diff
@@ -34,6 +34,12 @@
 (defmulti label-changer view-type-dispatcher)
 (defmulti action-buttons view-type-fn-dispatcher)
 
+;; MAP TABLE
+(defmulti map-table-page view-type-dispatcher)
+
+;; MAP TABLE
+(defmulti map-table-page view-type-dispatcher)
```
Comment (pld) [inline src/hatti/views.cljs]: this is on line 38, should be removed

Comment (pld) [inline src/hatti/views.cljs]: this is on line 38, should be removed

File: src/hatti/shared.cljs [hunk 4]
```diff
@@ -35,7 +35,9 @@
   "An initial, empty, app-state, which can be modified to change dataviews."
   []
   (atom
-   {:views {:all [:overview :map :table :chart :saved-charts :settings]
+   {:views {:all [:overview :map :table :chart :saved-charts :settings
+                  :map-table
```
Comment (pld) [inline src/hatti/shared.cljs]: if we don't add these to that atom can we merge this into master without changing the default functionality?

Comment (pld) [inline src/hatti/shared.cljs]: we talked about this on slack, right now Secretary will only be able to redirect to routes that are added here, so we can't actually remove them. Before merging to master we'll need a solution for this.

**Pull request closed without merging**